pkg/web_app/lib/src/widget/downloads_chart/widget.dart

```diff
@@ -6,34 +6,260 @@ import 'dart:convert';
+  // We add half of the line thickness at both ends of the x-axis so that it
+  // covers the vertical ticks at the end.
+  xaxis.setAttribute('d',
+      'M${xZero - (lineThickness / 2)} $yZero L${xMax + (lineThickness / 2)} $yZero');
+  chart.append(xaxis);
+
+  late SVGTextElement firstTickLabel;
+  // place a tick every 4 weeks
+  for (int week = 0; week < values.length; week += 4) {
+    final date = computeDateForWeekNumber(newestDate, values.length, week);
+    final (x, y) = computeCoordinates(date, 0);
+
+    final tick = SVGPathElement();
+    tick.setAttribute('class', 'downloads-chart-x-axis');
+    tick.setAttribute('d', 'M$x $y l0 $tickLength');
+    chart.append(tick);
+
+    final tickLabel = SVGTextElement();
+    chart.append(tickLabel);
+    tickLabel.setAttribute(
+        'class', 'downloads-chart-tick-label  downloads-chart-tick-label-x');
+    tickLabel.text = formatAbbrMonthDay(date);
+    tickLabel.setAttribute('y', '$tickLabelYCoordinate');
+    tickLabel.setAttribute('x', '$x');
+
+    if (week == 0) {
+      firstTickLabel = tickLabel;
+    }
+  }
+
+  for (int i = 0; i <= maxY / interval; i++) {
+    final (x, y) = computeCoordinates(firstDate, i * interval);
+
+    final tickLabel = SVGTextElement();
+    tickLabel.setAttribute(
+        'class', 'downloads-chart-tick-label  downloads-chart-tick-label-y');
+    tickLabel.text =
+        '${compactFormat(i * interval).value}${compactFormat(i * interval).suffix}';
+    tickLabel.setAttribute('x', '${xMax + padding}');
+    tickLabel.setAttribute('y', '$y');
+    chart.append(tickLabel);
+
+    if (i == 0) {
+      // No long tick in the bottom, we have the x-axis here.
+      continue;
+    }
+
+    final longTick = SVGPathElement();
+    longTick.setAttribute('class', 'downloads-chart-frame');
+    longTick.setAttribute('d',
+        'M${xZero - (lineThickness / 2)} $y L${xMax - (lineThickness / 2)} $y');
+    chart.append(longTick);
+  }
+
+  // We use the clipPath to cut the ends of the chart lines so that we don't
+  // draw outside the frame of the chart.
+  final clipPath = SVGClipPathElement();
+  clipPath.setAttribute('id', 'clipRect');
+  final clipRect = SVGRectElement();
+  clipRect.setAttribute('y', '$yMax');
+  clipRect.setAttribute('height', '${chartheight - (lineThickness / 2)}');
+  clipRect.setAttribute('x', '$xZero');
+  clipRect.setAttribute('width', '$chartWidth');
+  clipPath.append(clipRect);
+  chart.append(clipPath);
+
+  // Chart lines and legends
+
+  final lines = <StringBuffer>[];
+  for (int versionRange = 0; versionRange < values[0].length; versionRange++) {
+    final line = StringBuffer();
+    var c = 'M';
+    for (int week = 0; week < values.length; week++) {
+      final (x, y) = computeCoordinates(
+          computeDateForWeekNumber(newestDate, values.length, week),
+          values[week][versionRange]);
+      line.write(' $c$x $y');
+      c = 'L';
+    }
+    lines.add(line);
+  }
+
+  double legendXCoor = xZero;
```

Comment:
nit: avoid abbreviations

Either spell it out
```suggestion
  double legendXCoordinate = xZero;
```

or leave it (we know x is a coordinate)

```suggestion
  double legendX= xZero;
```

Reply:
done, also for y.